src/devices/Ads1115/Ads1115.cs

```diff
@@ -104,14 +220,255 @@ private void SetConfig()
+            catch (Exception)
+            {
+                gpioController.ClosePin(pin);
+                throw;
+            }
+
+            _gpioController = gpioController;
+            _alrtRdyPin = pin;
+        }
+
+        private void ConversionReadyCallback(object sender, PinValueChangedEventArgs pinValueChangedEventArgs)
+        {
+            if (AlertReadyAsserted != null)
+            {
+                AlertReadyAsserted();
+            }
+        }
+
+        /// <summary>
+        /// Enable comparator callback mode.
+        /// In traditional comparator mode, the callback is triggered each time the measured value exceeds the given upper value (for
+        /// the given queueLength number of samples). It deasserts when the lower value is reached.
+        /// In window comparator mode, the callback is triggered each time the measured value exceeds the given upper value or gets
+        /// less than the given lower value.
+        /// </summary>
+        /// <param name="lowerValue">Lower value for the comparator</param>
+        /// <param name="upperValue">Upper value for the comparator</param>
+        /// <param name="mode">Traditional or Window comparator mode</param>
+        /// <param name="queueLength">Minimum number of samples that must exceed the threshold to trigger the event</param>
+        /// <param name="gpioController">The GPIO controller, to configure the pin</param>
+        /// <param name="pin">The pin where the ALRT/RDY pin is connected to. </param>
+        public void EnableComparator(double lowerValue, double upperValue, ComparatorMode mode,
+            ComparatorQueue queueLength, GpioController gpioController, int pin)
+        {
+            EnableComparator(VoltageToRaw(lowerValue), VoltageToRaw(upperValue), mode, queueLength, gpioController, pin);
+        }
+
+        /// <summary>
+        /// Enable comparator callback mode.
+        /// In traditional comparator mode, the callback is triggered each time the measured value exceeds the given upper value (for
+        /// the given queueLength number of samples). It deasserts when the lower value is reached.
+        /// In window comparator mode, the callback is triggered each time the measured value exceeds the given upper value or gets
+        /// less than the given lower value.
+        /// </summary>
+        /// <param name="lowerValue">Lower value for the comparator</param>
+        /// <param name="upperValue">Upper value for the comparator</param>
+        /// <param name="mode">Traditional or Window comparator mode</param>
+        /// <param name="queueLength">Minimum number of samples that must exceed the threshold to trigger the event</param>
+        /// <param name="gpioController">The GPIO controller, to configure the pin</param>
+        /// <param name="pin">The pin where the ALRT/RDY pin is connected to. </param>
+        public void EnableComparator(short lowerValue, short upperValue, ComparatorMode mode,
```

Comment:
does it make sense to make this private? how would user know what is lower and upper value without reading the spec? I'm fine with exposing register as property on top of that but take uint/int instead of two shorts.

Reply:
It would certainly simplify the interface. However, we do offer the ReadRaw() method as well, which returns the same unit (and if you use that, you have to know how to convert the value).